sql/core/src/test/scala/org/apache/spark/sql/connector/ProcedureSuite.scala

```diff
@@ -40,15 +40,23 @@ class ProcedureSuite extends QueryTest with SharedSparkSession with BeforeAndAft
 
   before {
     spark.conf.set(s"spark.sql.catalog.cat", classOf[InMemoryCatalog].getName)
+    spark.conf.set(s"spark.sql.catalog.cat2", classOf[InMemoryCatalog].getName)
+
+    // needed for switching back and forth between catalogs
+    sql("create database cat.default")
```

Comment:
`default` is a special name, shall we use common names like `my_db1`?

Reply:
This actually to fix an error .  These catalogs don't have a default database.  When I run 'use cat', it will complain about that.  Any better way?

Reply:
Specifically:

```
[SCHEMA_NOT_FOUND] The schema `cat`.`default` cannot be found. Verify the spelling and correctness of the schema and catalog.
If you did not qualify the name with a catalog, verify the current_schema() output, or qualify the name with the correct catalog.
```

Reply:
The default namespace for these catalogs should be `[]`.

Reply:
Yea found it, this is because of https://github.com/apache/spark/blob/master/sql/core/src/test/scala/org/apache/spark/sql/test/SQLTestUtils.scala#L383  , the test fixture does 'use default' at the end.  Changed to use 'withNamepsace'